src/app/features/tech-record/components/tech-record-amend-vrm/tech-record-amend-vrm.component.ts

```diff
@@ -0,0 +1,153 @@
+import cloneDeep from 'lodash.clonedeep';
+import { catchError, map, of, take, throwError } from 'rxjs';
+import { ValidatorNames } from '@forms/models/validators.enum';
+import { Actions, ofType } from '@ngrx/effects';
+
+@Component({
+  selector: 'app-change-amend-vrm',
+  templateUrl: './tech-record-amend-vrm.component.html',
+  styleUrls: ['./tech-record-amend-vrm.component.scss']
+})
+export class AmendVrmComponent implements OnInit, OnChanges {
+  vehicle?: VehicleTechRecordModel;
+  currentTechRecord?: TechRecordModel;
+  form: CustomFormGroup;
+  width: FormNodeWidth = FormNodeWidth.L;
+
+  template: FormNode = {
+    name: 'criteria',
+    type: FormNodeTypes.GROUP,
+    children: [
+      {
+        name: 'newVrm',
+        label: 'Input a new VRM',
+        value: '',
+        type: FormNodeTypes.CONTROL,
+        validators: [{ name: ValidatorNames.MaxLength, args: 9 }, { name: ValidatorNames.MinLength, args: 1 }, { name: ValidatorNames.Alphanumeric }]
+      }
+    ]
+  };
+
+  constructor(
+    private actions$: Actions,
+    public dfs: DynamicFormService,
+    private globalErrorService: GlobalErrorService,
+    private route: ActivatedRoute,
+    private router: Router,
+    private store: Store<TechnicalRecordServiceState>,
+    private technicalRecordService: TechnicalRecordService
+  ) {
+    this.technicalRecordService.selectedVehicleTechRecord$.pipe(take(1)).subscribe(vehicle => (this.vehicle = vehicle));
+
+    this.technicalRecordService.editableTechRecord$.pipe(take(1)).subscribe(techRecord => (this.currentTechRecord = techRecord));
+
+    this.form = this.dfs.createForm(this.template) as CustomFormGroup;
+  }
+
+  ngOnInit(): void {
+    if (!this.currentTechRecord) {
+      this.navigateBack();
+    }
+
+    this.actions$.pipe(ofType(updateTechRecordsSuccess), take(1)).subscribe(() => {
+      this.navigateBack();
+    });
+  }
+
+  ngOnChanges(): void {
+    this.globalErrorService.clearErrors();
+  }
+
+  get makeAndModel(): string {
+    const c = this.currentTechRecord;
+    if (!c?.make && !c?.chassisMake) return '';
+
+    return `${c.vehicleType === 'psv' ? c.chassisMake : c.make} - ${c.vehicleType === 'psv' ? c.chassisModel : c.model}`;
+  }
+
+  get vrm(): string | undefined {
+    return this.vehicle?.vrms.find(vrm => vrm.isPrimary === true)?.vrm;
+  }
+
+  navigateBack() {
+    this.globalErrorService.clearErrors();
+    this.router.navigate(['..'], { relativeTo: this.route });
+  }
+
+  handleSubmit(newVrm: string): void {
+    this.globalErrorService.clearErrors();
+    if (newVrm === '' || (newVrm === this.vrm ?? '')) {
+      return this.globalErrorService.addError({ error: 'You must provide a new VRM', anchorLink: 'newVrm' });
+    }
+
+    this.technicalRecordService
+      .isUnique(newVrm, SEARCH_TYPES.VRM)
+      .pipe(
+        map(response => {
+          console.log(response);
+          return response;
+        }),
```

Comment:
This `map()` is not doing anything. If the `console.log(response)` is intentional, please change it to a `tap()`.
```suggestion
        tap(response => console.log(response)),
```